uat/custom-components/src/main/java/com/aws/greengrass/artifacts/ShadowReactiveComponent.java

```diff
@@ -97,83 +90,45 @@ private boolean isNullOrEmpty(String s) {
-        UpdateStateRequest updateStateRequest = new UpdateStateRequest();
-        updateStateRequest.setState(state);
-        greengrassCoreIPCClient.updateState(updateStateRequest, Optional.empty()).getResponse().get();
-    }
-
     private void subscribeToShadowUpdates(String thingName, String shadowName, String topic, JsonNode expectedUpdateShadowDocumentJson, JsonNode expectedUpdateShadowDocumentJson2) {
+        SubscribeToTopicRequest subscribe = new SubscribeToTopicRequest().withTopic(topic);
         try {
-            SubscribeToTopicRequest subscribe = new SubscribeToTopicRequest();
-            subscribe.setTopic(topic);
-            eventStreamRpcConnection = IPCTestUtils.getGreengrassClient();
-            GreengrassCoreIPC greengrassCoreIPCClient = eventStreamRpcConnection.getClient();
-            CompletableFuture<SubscribeToTopicResponse> fut = greengrassCoreIPCClient.subscribeToTopic(subscribe, Optional.of(new StreamResponseHandler<SubscriptionResponseMessage>() {
-                @Override
-                public void onStreamEvent(SubscriptionResponseMessage message) {
-                    String payload = new String(message.getBinaryMessage().getMessage(), UTF_8);
-                    LOGGER.info("Received new message: {}", payload);
-                    try {
-                        LOGGER.info("Received new message on {}", topic);
-                        if (topic.contains("delta")) {
-                            JsonNode retrievedDeltaPayloadJson = MAPPER.readTree(message.getBinaryMessage().getMessage());
-                            JsonNode newUpdateDocument = calculateNewDocument(retrievedDeltaPayloadJson, expectedUpdateShadowDocumentJson);
-                            handleUpdateThingShadowOperation(thingName, shadowName, MAPPER.writeValueAsBytes(newUpdateDocument), expectedUpdateShadowDocumentJson2);
+            greengrassCoreIPCClient.subscribeToTopic(subscribe, (message) ->
+                    {
+                        String payload = new String(message.getBinaryMessage().getMessage(), UTF_8);
+                        LOGGER.info("Received new message: {}", payload);
+                        try {
+                            LOGGER.info("Received new message on {}", topic);
+                            if (topic.contains("delta")) {
+                                JsonNode retrievedDeltaPayloadJson = MAPPER
+                                        .readTree(message.getBinaryMessage().getMessage());
+                                JsonNode newUpdateDocument = calculateNewDocument(retrievedDeltaPayloadJson, expectedUpdateShadowDocumentJson);
+                                handleUpdateThingShadowOperation(thingName, shadowName, MAPPER
+                                        .writeValueAsBytes(newUpdateDocument), expectedUpdateShadowDocumentJson2);
+                            }
+                        } catch (IOException | ExecutionException | InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)